app/models/user.rb

```diff
@@ -659,6 +659,20 @@ def index_observations
   def merge(reject)
     raise "Can't merge a user with itself" if reject.id == id
     reject.friendships.where(friend_id: id).each{ |f| f.destroy }
+
+    # Conditions should match index_votes_on_unique_obs_fave
+    reject.votes
+          .joins( "JOIN votes AS keeper_votes USING (votable_type, votable_id)" )
+          .where(
+            keeper_votes: {
+              voter_type: self.class.polymorphic_name,
+              voter_id: id,
+              vote_scope: nil
+            },
+            votable_type: Observation.polymorphic_name,
+            vote_scope: nil
+          )
+          .destroy_all
```

Comment:
This does concern me a bit. I can think of one user with at least 55,000 faves, so that would cause problems here (though so would attempting to delete that user). On the other hand, I think you'd need to reindex all of the observations these votes are associated with if you used `delete_all`. Any thoughts on this, @pleary?

Reply:
This is destroying just the duplicates. I don't believe I have a way to check, but it seems unlikely that two user records have more than 100 faves in common.

Reply:
I agree that there would be a potential performance problem if the two users had a lot of faves in common, but also that is unlikely to be the case. I tested the query against two users with the most votes, where none were in common, and it was quite fast. If we think that case could pop up, then we should turn this into a delayed job. But I think it's unlikely enough that I'm OK with the code as-is.